src/types/transaction.zig

```diff
@@ -145,12 +217,37 @@ test "Transaction basics" {
         try tx.addOutput(50000, script_pubkey);
     }
 
-    try testing.expectEqual(@as(usize, 1), tx.inputs.items.len);
-    try testing.expectEqual(@as(usize, 1), tx.outputs.items.len);
-    try testing.expectEqual(@as(i64, 50000), tx.outputs.items[0].value);
+    try testing.expectEqual(@as(usize, 1), tx.inputs.len);
+    try testing.expectEqual(@as(usize, 1), tx.outputs.len);
+    try testing.expectEqual(@as(i64, 50000), tx.outputs[0].value);
 
     _ = tx.hash();
 
     const vsize = tx.virtual_size();
     try testing.expect(vsize > 0);
 }
+
+test "Transaction serialization" {
+    const testing = std.testing;
+    const allocator = testing.allocator;
+
+    var tx = try Self.init(allocator);
+    defer tx.deinit();
+
+    try tx.addInput(OutPoint{ .hash = Hash.zero(), .index = 0 });
+
+    {
+        var script_pubkey = try Script.init(allocator);
+        defer script_pubkey.deinit();
+        try script_pubkey.push(&[_]u8{ 0x76, 0xa9, 0x14 }); // OP_DUP OP_HASH160 Push14
```

Comment:
use the Opcode enum rather than hex values

Reply:
Same as previous

Reply:
same, keep improving the codebase